dts/bindings/pinctrl/nordic,nrf-pinctrl.yaml

```diff
@@ -119,3 +119,9 @@ child-binding:
         description: |
           Invert pin polarity (set the active state to low).
           Only valid for PWM channel output pins.
+
+      nordic,clock-enable:
+        type: boolean
+        description: |
+          Enable clock for the pin. Only valid or required for certain
+          peripherals, refer to the reference manual.
```

Comment:
I understand your concerns, you're totally right. But I guess it's similar if you e.g. forget to configure I2C pins in open-drain mode, or that you need to specify the "high-drive" mode when operating at certain speeds. New devices will actually become more complicated, as the "route wherever you want" does not apply anymore (the low speed peripheral world is over). I wish a more standard pinmux technique had been implemented, this clock setting is confusing. This setting will be documented in the reference manual somehow, so users will need to understand this when setting up their board pinctrl. Note that this PR changes a few overlays because there are no "default" SPI pins on the DK (not sure why), whereas this does not happen for e.g. uart, can, etc.

Reply:
We could add predefined autogenerated pinctrl groups for the peripherals which can only be routed to fixed IOs, similar to what `st` does with their hal https://github.com/zephyrproject-rtos/hal_stm32/blob/main/dts/st/h5/stm32h503cbtx-pinctrl.dtsi , we can add these fun extra configurations like `nordic,clock-enable` to those like:

```
/omit-if-no-ref/ spi130_default: spi130_default {
	group1 {
		psels = <NRF_PSEL(SPIM_SCK, 0, 0)>,
			<NRF_PSEL(SPIM_MISO, 0, 6)>,
			<NRF_PSEL(SPIM_MOSI, 0, 7)>;
		nordic,clock-enable;
	};

	group2 {
		psels = <NRF_PSEL(SPIM_MISO, 0, 6)>;
	};
};
```
and have the user simply add
```
&spi130 {
	pinctrl-0 = <&spi130_default_alt>;
	pinctrl-1 = <&spi130_sleep_alt>;
	pinctrl-names = "default", "sleep";
	...
};
```
Not sure SPI is the best example, TWI is probably better :)

Reply:
I will say that it is way beyond the scope of this PR though, I see no better simple alternative to the dts flag :)

Reply:
> it's similar if you e.g. forget to configure I2C pins in open-drain mode

I don't think so. open-drain is something that i2c user may expect and clock-enable is something totally unexpected to the user. It's difficult to explain to the user why it is needed and even worst which instance needs it. Imo this should be property of a node (e.g. `uart136 { clock-pin; }` in nrf54h20.dtsi). Though, it might be hard to transfer that from node to pinctrl. I only came with an idea where during the initialization you create pinctrl config copy on stack and patch the configuration. Not sure if that is violating pinctrl api.

Reply:
What about cases where a certain drive mode is needed to operate at high speed? You also need the reference manual to know this is required. I agree it's not ideal, but this setting being instance dependent really made me think it's the "less bad solution". This may be improved in future SoCs to make software easier @hubertmis

Reply:
Another thing I've found: on I3C, unless you configure the `nordic,drive-mode = <NRF_DRIVE_H0D1>;`, waveforms do not keep up (due to its _high-speed_ I guess).